.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,153 @@
+name: MCP E2E Tests
```

Comment:
Can you generate a README file for the test dir to describe what the test strategy is, how it works, how to run it, what it checking.